docs/code/CrossPlatform.md

```diff
@@ -0,0 +1,157 @@
+		- [2.2 Build](#22-build)
+		- [2.3 Managed Code](#23-managed-code)
+		- [2.4 Native Projects](#24-native-projects)
+		- [2.5 3rd Party Dependencies](#25-3rd-party-dependencies)
+	- [3 Possible Solutions](#3-possible-solutions)
+		- [3.1 Fully Remove Intel MKL](#31-fully-remove-intel-mkl)
+		- [3.2 Find Replacements as needed](#32-find-replacements-as-needed)
+		- [3.3 Software fallbacks](#33-software-fallbacks)
+		- [3.4 Hybrid Between Finding Replacements and Software Fallbacks](#34-hybrid-between-finding-replacements-and-software-fallbacks)
+	- [My Suggestion](#my-suggestion)
+		- [NET Core 5](#net-core-5)
+		- [3rd Party Dependencies](#3rd-party-dependencies)
+		- [Helix](#helix)
+		- [Mobile Support](#mobile-support)
+		- [Support Grid](#support-grid)
+
+## 1. Why cross-platform/architecture
+ML.NET is an open-source machine learning framework which makes machine learning accessible to .NET developers with the same code that powers machine learning across many Microsoft products, including Power BI, Windows Defender, and Azure.
+
+ML.NET allows .NET developers to develop/train their own models and infuse custom machine learning into their applications using .NET.
+
+Currently, while .NET is able to run on many different platforms and architectures, ML.NET is only able to run on Windows, Mac, and Linux, either x86 or x64. This excludes many architectures such as Arm, Arm64, M1, and web assembly, places that .NET is currently able to run.
+
+The goal is to enable ML.NET to run everywhere that .NET itself is able to run.
+
+## 2. Current status
+### 2.1 Problems
+ML.NET has a hard dependency on Intel MKL. It performs many optimized math functions and enables several transformers/trainers to be run natively for improved performance. The problem is that Intel MKL can only run on x86/x64 machines and is the main blocker for expanding to other architectures.
+
+ML.NET also has dependencies on things that either don't build on other architectures or have to be compiled by the user if its wanted. For example:
+ - LightGBM
+ - TensorFlow
+ - OnnxRuntime
+
+I will go over these in more depth below.
+
+### 2.2 Build
+Since ML.NET has a hard dependency on Intel MKL, the build process assumes it's always there. For example, the build process will try and copy dlls without checking if they exist. The build process will need to be modified so that it doesn't fail when it can't find these files. It does the same copy check for our own Native dlls, so this will need to be fixed for those as well.
+
+### 2.3 Managed Code
+Since ML.NET has a hard dependency on Intel MKL, the managed code imports the dlls without checking whether or not they exist. If the dlls don't exist you get a hard failure. For example, the base test class imports and sets up Intel MKL even if the test itself does not need it. This same situation applies to our own native dlls. When they aren't present, the imports fail. We will need to make ML.NET correctly handle the dll imports.
```

Comment:
You mention tests here, what about the product?

Reply:
The product does this as well. I was just trying to give an example of how the current behavior is regarding Intel MKL and native dependencies. Do you think I should make it more clear that it does in the product as well?

Reply:
Yes, the product problem is the first order issue.  The fact that the product does this means that our software fallback paths are somewhat "dead code" right now, so we there may be some additional cost involved in enabling them to ensure they function correctly.

Reply:
We intentionally factored out any components that depend on MKL into the `Microsoft.ML.Mkl.Components` library. It makes sense to me that since these components are in "MKL.Components" that they assume MKL is available.